app/src/main/java/com/everest/movieapp/data/repository/MovieRepository.kt

```diff
@@ -0,0 +1,74 @@
+package com.everest.movieapp.data.repository
+
+import android.content.Context
+import android.net.ConnectivityManager
+import com.everest.movieapp.data.api.ApplicationContextProvider
+import com.everest.movieapp.data.api.MovieApi
+import com.everest.movieapp.data.model.Result
+import com.everest.movieapp.data.model.UiMovieDetails
+import com.everest.movieapp.data.room.MovieRoomDataBase
+import com.everest.movieapp.utils.constants.Constants.IMAGE_BASE_URL
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+
+class MovieRepository(private val movieApi: MovieApi) {
+    private val context = ApplicationContextProvider.getInstance()
+    private var movieRoomDataBase = MovieRoomDataBase.getDatabase(context?.applicationContext!!)
+
+    suspend fun getPopularMovies() = flow {
```

Comment:
This is not returning a stream of response right, instead only returning single value. Check if using flow here is necessary or overkill